src/users/src/users-service/handlers.go

```diff
@@ -206,18 +206,35 @@ func CreateEndpointAndSendConfirmation(updateEndpointInput *pinpoint.UpdateEndpo
 		sendMessageAddress[phonenumber] = &pinpoint.AddressConfiguration {
 			ChannelType: aws.String("SMS"),
 		}
-		sendMessageInput := &pinpoint.SendMessagesInput {
-			ApplicationId: &pinpoint_app_id,
-			MessageRequest: &pinpoint.MessageRequest {
-				Addresses: sendMessageAddress,
-				MessageConfiguration: &pinpoint.DirectMessageConfiguration {
-					SMSMessage: &pinpoint.SMSMessage {
-						Body: aws.String("Reply Y to receive one time automated marketing messages at this number. No purchase necessary. T&C apply."),
-						MessageType: aws.String("TRANSACTIONAL"),
-						OriginationNumber: &pinpoint_sms_long_code,
+		var sendMessageInput *pinpoint.SendMessagesInput 
+		if (pinpoint_sms_long_code == "NONE") {
+			fmt.Println("Long code not assigned by user");
+			sendMessageInput = &pinpoint.SendMessagesInput {
+				ApplicationId: &pinpoint_app_id,
+				MessageRequest: &pinpoint.MessageRequest {
+					Addresses: sendMessageAddress,
+					MessageConfiguration: &pinpoint.DirectMessageConfiguration {
+						SMSMessage: &pinpoint.SMSMessage {
+							Body: aws.String("Reply Y to receive one time automated marketing messages at this number. No purchase necessary. T&C apply."),
+							MessageType: aws.String("TRANSACTIONAL"),
+						},
```

Comment:
So a long code is not required for two-way SMS? The only difference between the "if" and "else" block that I can see is that `OriginationNumber` is included in the else block. Will two-way still work without it and if so why are we dealing with long codes at all?

Reply:
So the thing is, we are having 2-way SMS. If we dont set up the long code and enable two way SMS for it, customers will receive the confirmation message asking them to  opt in, but when the customers reply with a Y, they won't get any response. Pinpoint does assign a random number to send messages if you dont assign one, it is just for the 2 way SMS that we have to particularly obtain one.  Once you assign the long code, pinpoint won't randomly assign any long/short code, it would pick up the one you have.

Reply:
So if I understand correctly, the 2-way SMS UX that we're designing here will only work if a long code has been setup, right? If that is the case, should we been sending the opt-in SMS if there is no long code? Won't that create a confusing UX since the customer's reply won't be received and processed as expected?

Reply:
Added the long code ssm as well as CFN as discussed offline.